raster-foundry/raster-foundry
# Make async jobs terminate

## Overview

This PR uses `cats-effect` `Resource`s for thread pool and connection pool management and ends `run` methods with `System.exit`.

It also removes two jobs from export that don't appear to be used anywhere.

### Checklist

- [x] Landsat 8 import
- [x] Sentinel-2 import
- [x] Create export def
- [x] Export
- [x] Update export status
- [x] Update AOI Project
- [x] Find AOI project
- ~S3 copy~
- ~Dropbox copy~

### Notes

Unclear why this became necessary :man_shrugging:

## Testing Instructions

 * count the scenes in your db by hand or with sql
 * reassemble batch jar
 * `./scripts/console batch bash`
 * `java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main import_landsat8_c1 2018-09-29` <-- some date in the past doesn't really matter
 * wait for it to finish (it will, I swear)
 * count scenes again there should be more
 * Also do an export -- it should succeed or fail but not run forever
 * Others are optional because changes were way less dramatic
   * `rf find-aoi-projects`
   * `rf update-aoi-project <a project id>` (might fail for missing query parameter json decoder if it's actually doing any work, in which case, you should rebase on #4267 and try again )
   * *NOT RECOMMENDED because so many scenes and requester pays* `java -cp /opt/raster-foundry/jars/batch-assembly.jar com.azavea.rf.batch.Main import_sentinel2 <a date>`

Closes azavea/raster-foundry-platform#524
